### PR TITLE
Fix various CNI add/delete handling bugs

### PR DESCRIPTION
**Pull request by tgraf**
Depends on #7667

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7670)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
The CI failure here was real, the CNI binary appeared right in between a ADD and DELETE request which caused the DELETE to be handled via the cilium-cni binary but the endpoint did not exist. This caused kubelet to retry forever without ever re-creating the pod. Requires a fix.

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: plugins/cilium-cni/cilium-cni.go [hunk 1]
```diff
@@ -622,36 +622,49 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 }
 
 func cmdDel(args *skel.CmdArgs) error {
-	log.WithField("args", args).Debug("Processing CNI DEL request")
+	// Note about when to return errors: kubelet will retry the deletion
+	// for a long time. Therefore, only return an error for errors which
+	// are guaranteed to be recoverable.
+	log.WithField("args", args).Debugf("Processing CNI DEL request %+v", args)
```
Comment (ianvernon) [inline plugins/cilium-cni/cilium-cni.go]: Won't args be printed twice here? I don't see a need for this log change.

File: plugins/cilium-cni/cilium-cni.go [hunk 2]
```diff
@@ -622,36 +622,49 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
...
+		log.WithError(err).Warningf("Unable to enter namespace %q, will not delete interface", args.Netns)
+		// We are not returning an error as this is very unlikely to be recoverable
+		return nil
 	}
 	defer netNs.Close()
 
-	return netns.RemoveIfFromNetNSIfExists(netNs, args.IfName)
+	err = netns.RemoveIfFromNetNSIfExists(netNs, args.IfName)
+	if err != nil {
+		log.WithError(err).Warningf("Unable to delet interface %s in namespace %q, will not delete interface", args.IfName, args.Netns)
```
Comment (ianvernon) [inline plugins/cilium-cni/cilium-cni.go]: delet --> delete

**Comment by ianvernon** [comment]
Should we backport this to `v1.3` and `v1.4`?

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**